sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/CHANGELOG.md

```diff
@@ -4,6 +4,9 @@
 
 ### Features Added
 
+* Added service version and custom attributes specified on the resource object to the published telemetry.
```

Comment:
Please add a more detailed description of the change. Its not clear if this is applicable to all signals (logs, metrics and traces). 

If this change is taking every resource attribute and attaching them to logs, traces and metrics, then this should be designed first. Resource could contain potentially high cardinality items, and attaching them to Metric properties can exhaust the metric time series limit. 

I suggest to have an issue first, where the detailed design is described.

Reply:
Will add the clarification as it applies to all three. It's my understanding that by design, the resource object in OpenTelemetry is expected to be associated with any type of telemetry emitted (it represents the source of the telemetry and is a place to capture things identifying the service, its environment, version, namespace, etc.).  If someone can clarify if it is expected on metrics or not, that would be great. It's very easy to back that out.  Can you clarify what you mean by "high cardinality items"?  Right now, I think the implementation is only expecting strings (which may be an incorrect assumption).

Reply:
Here's the OpenTelemetry documentation for resources:  https://opentelemetry.io/docs/instrumentation/net/resources/.  Based on some of the information there, it could be used for logs, traces, or metrics.

Reply:
From their glossary:

**Resource**
Captures information about the entity for which telemetry is recorded. For example, a process producing telemetry that is running in a container on Kubernetes has a pod name, it is in a namespace and possibly is part of a deployment which also has a name. All three of these attributes can be included in the Resource and applied to any data source.

Reply:
I understand Resource in OpenTelemetry is applicable to all signals. My question is about applying it specifically for Azure Monitor. It should be an opt-in feature and/or should be flexible enough for users to pick which resource attributes they want attached to their telemetry, instead of attaching everything by default. (For Metrics, it could eat up cardinality, and for all signals, it adds up ingestion-size and can increase Azure bill for users)

There should be a design first on how AzureMonitor want to model Resources (attach to all telemetry or store in one place, and refer to it from other telemetry, how to opt-in/out, how to customize etc.)

Reply:
@cijothomas would it be ok to just add `service.version` as there is a specific AppInisghts property for it? It sounds arbitrary resource attributes should not be added because it's not yet clear how it should be done - is it correct?

Reply:
In support of @cijothomas suggestion, I've updated the PR with a boolean flag configuration option (in AzureMonitorExporterOptions) that will enable/disable sending custom attributes.  It defaults to false, so the behavior is by default the current baseline behavior (with the exception of service.version which would always be sent).

Reply:
Another thought if the flexibility was desired to selectively filter the custom attributes, a custom action could be added to the AzureMonitorExporterOptions that could be called to see if a given attribute should be included in telemetry or not (maybe a future PR for that one).

Reply:
> I understand Resource in OpenTelemetry is applicable to all signals. My question is about applying it specifically for Azure Monitor. It should be an opt-in feature and/or should be flexible enough for users to pick which resource attributes they want attached to their telemetry, instead of attaching everything by default. (For Metrics, it could eat up cardinality, and for all signals, it adds up ingestion-size and can increase Azure bill for users)
> 
> There should be a design first on how AzureMonitor want to model Resources (attach to all telemetry or store in one place, and refer to it from other telemetry, how to opt-in/out, how to customize etc.)

Can you elaborate on the "store in one place and refer to it from other telemetry".  How does that work with the ingestion on App Insights and get displayed when looking at transactions in the portal?  Also, curious how this defers from other exporters (OTLP, Geneva, etc.).

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/36063 This PR just got merged. I am not sure if it'll have the details on "how", but I expect backend/UI to understand resources via the mechanism from that PR.

Reply:
@treyjones-ms https://github.com/Azure/azure-sdk-for-net/pull/36063 has an updated description, could you please check and let us know if you have any questions.

Reply:
@rajkumar-rangaraj - I put a comment on your PR (even though it's already merged). I'm guessing my PR might be OBE at this point, but confirm after reviewing my comment.

Reply:
Other pull requests address the part of the issue that you're attempting to cover. Instead of duplicating the resource attributes in every trace and log, having a common storage of resource with a join would be beneficial.